### DISCUSSION TITLE
Using BERTopic in a for loop throwing error

### DISCUSSION THREAD

**Discussion by mbhatley** [MaartenGr/BERTopic#1350] [Q&A] [answered]
Hi,

I am currently getting a `ValueError: Transform unavailable when model was fit with only a single data sample.`  when trying to use BERTopic in a for loop. 

The elements (i.e. `words`) in the loop is an entire article. I believe the issue in the `vector` part, but I cannot seem to get a solution here. 

For reference, the code is:

```
keyphrase_topic_model = BERTopic(vectorizer_model=KeyphraseCountVectorizer())

keywords,contents = [], []

for words in df['content']:
    content = words
    vector = [words]
    keyphrase = keyphrase_topic_model.fit_transform(vector)
    keyphrases = [phrase[0] for phrase in keyphrase]
    
    contents.append(content)
    keywords.append(keyphrase_topics)
    
    df5 = pd.DataFrame({"content": contents, "keywords": keywords})
```

**Comment by MaartenGr** [comment, marked as answer]
The `fit` function is not meant to be used in a loop as it will overwrite the previous fit. Moreover, topic modelong does not work with a single document. You would have to use something like KeyBERT instead: https://github.com/MaartenGr/KeyBERT

**Reply by mbhatley** [reply]
Hi @MaartenGr , thank you! KeyBERT working well in a loop.
